cpp/references/data_structures/data_exporter.cpp

```diff
@@ -0,0 +1,57 @@
+#include "data_exporter.hpp"
+
+using namespace godot;
+
+Variant IDataExporter::_get_data() const { return Variant(); }
+
+void IDataExporter::_set_data(Variant) {}
+
+Variant IDataExporter::_get(String p_property) {
+	Variant vdata = _get_data();
+	ERR_FAIL_COND_V(vdata.get_type() != Variant::OBJECT, false);
+
+	Ref<Reference> instance = vdata;
+	ERR_FAIL_COND_V(instance.is_valid(), false);
+
+	return instance->get(p_property);
+}
+
+bool IDataExporter::_set(String p_property, Variant p_value) {
+	Variant vdata = _get_data();
+	ERR_FAIL_COND_V(vdata.get_type() != Variant::OBJECT, false);
+
+	Ref<Reference> instance = vdata;
+	ERR_FAIL_COND_V(instance.is_valid(), false);
+
+	instance->set(p_property, p_value);
+	return true;
+}
+
+Array IDataExporter::_get_property_list() {
+	Array ret;
+	Variant vdata = _get_data();
+	ERR_FAIL_COND_V(vdata.get_type() != Variant::OBJECT, ret);
+
+	Ref<Reference> instance = vdata;
+	ERR_FAIL_COND_V(instance.is_valid(), ret);
+
+	Array all = instance->_get_property_list();
+	for (int i = all.size() - 1; i >= 0; --i) {
+		Dictionary pinfo = all[i].operator godot::Dictionary();
+		if (!(pinfo["usage"].operator int() & GODOT_PROPERTY_USAGE_SCRIPT_VARIABLE))
+			break;
+		ret.append(pinfo);
+	}
+	return ret;
+}
+
+void IDataExporter::_register_methods() {
```

Comment:
The IDataExporter class, as an interface class that is not instantiable, probably shouldn't be a "Godot" class with a `_register_methods()` function.